src/parser/jobs/sch/modules/Weaving.js

```diff
@@ -17,75 +16,32 @@ const WEAVING_SEVERITY = {
 }
 
 export default class Weaving extends CoreWeaving {
-	static dependencies = [
-		...CoreWeaving.dependencies,
-		'combatants',
-		'suggestions',
-	]
-	static title = t('sch.weaving.title')`Weaving Issues`
 	static displayOrder = DISPLAY_ORDER.WEAVING
 
-	constructor(...args) {
-		super(...args)
-		this.addHook('cast', {by: 'player'}, this._onScholarCast)
-	}
-
-	// Renamed so as not to overwrite parent module
-	// Just for tracking pos
-	_onScholarCast(event) {
-		const action = getDataBy(ACTIONS, 'id', event.ability.guid)
-		if (!action || !action.onGcd) {
-			return
+	// override to give SCH-relevant weave suggestions
+	// Pretty much copy-pasted from the CoreWeaving Module otherwise
+	_onComplete() {
+		// If there's been at least one gcd, run a cleanup on any remnant data
+		if (this._leadingGcdEvent) {
+			this._saveIfBad()
 		}
 
-		this._pos = this.combatants.selected.resources
-	}
-
-	// Now this, we want to overwrite
-	_onComplete() {
+		// Few triples is medium, any more is major
+		const badWeaves = this._badWeaves
```

Comment:
Any reason for copying the object like this? If you were filtering or something I get it but this is just a raw copy. Probably super minor in terms of performance overhead but yeah. Just a nit. You use the value only twice, and for the same thing, so maybe just assign like in this suggestion:

```suggestion
		const badWeaveCount = this._badWeaves.length
```

Reply:
No reason, actually. I have no problems changing it. One sec, and I'll update the PR.